crates/turbopack-node/js/src/ipc/error.ts

```diff
@@ -0,0 +1,58 @@
+// merged from next.js
+// https://github.com/vercel/next.js/blob/e657741b9908cf0044aaef959c0c4defb19ed6d8/packages/next/src/lib/is-error.ts
+// https://github.com/vercel/next.js/blob/e657741b9908cf0044aaef959c0c4defb19ed6d8/packages/next/src/shared/lib/is-plain-object.ts
+
+export default function isError(err: unknown): err is Error {
+  return (
+    typeof err === 'object' && err !== null && 'name' in err && 'message' in err
+  )
+}
+
+export function getProperError(err: unknown): Error {
+  if (isError(err)) {
+    return err
+  }
+
+  if (process.env.NODE_ENV === 'development') {
+    // provide better error for case where `throw undefined`
+    // is called in development
+    if (typeof err === 'undefined') {
+      return new Error(
+        'An undefined error was thrown, ' +
+        'see here for more info: https://nextjs.org/docs/messages/threw-undefined'
```

Comment:
Should these link to turbopack docs instead?

Reply:
hmm, ideally yes, maybe I'll just remove the doc link